### PR TITLE
BUG: Make reader.get_fields also return dropdowns with options

### LINKED ISSUE #391: PdfReader.get_form_text_fields() not returning dropdown fields
I have a pdf form with multiple Form-Fields in it. The function "getFormTextFields()" is only returning a dictionary without the dropdown-fields. The dictionary contains the information of all other fields, but the dropdown-fields are simply missing.

## Minimal Example

```python
from PyPDF2 import PdfReader

reader = PdfReader("sample-files/012-libreoffice-form/libreoffice-form.pdf")
reader.get_form_text_fields()
```

gives

```json
{"First Name": "Alice", "Last Name": "", "Birthday": "", "First Name_2": "Bob"}
```

The "Nationality" part is missing.

On the other hand:

```pycon
>>> reader.get_fields()
{'First Name': {'/FT': '/Tx', '/T': 'First Name', '/V': 'Alice', '/DV': 'Alice'},
 'Last Name': {'/FT': '/Tx', '/T': 'Last Name', '/V': '', '/DV': ''}, 
'female': {'/FT': '/Btn', '/Kids': [IndirectObject(7, 0), IndirectObject(9, 0)], '/T': 'female', '/Ff': 49152, '/V': '/Off', '/DV': '/Off'},
 'Birthday': {'/FT': '/Tx', '/T': 'Birthday', '/V': '', '/DV': ''}, 
'gdpr': {'/FT': '/Btn', '/T': 'gdpr', '/V': '/Off', '/DV': '/Off'}, 
'other': {'/FT': '/Btn', '/T': 'other', '/V': '/Off', '/DV': '/Off'}, 
'First Name_2': {'/FT': '/Tx', '/T': 'First Name_2', '/Ff': 4096, '/V': 'Bob', '/DV': 'Bob'}, 
'Nationality': {'/FT': '/Ch', '/T': 'Nationality', '/Ff': 131072, '/V': '', '/DV': ''}}
```

### PR DESCRIPTION
**Pull request by KourFrost**
Adding `/Opt` to the attributes list get_fields() will now add /Opt if it has it. For Combo Box /Ch it will list all of the available options in the drop down menu.  

Closes #391

### CODE REVIEW

File: PyPDF2/constants.py [hunk 1]
```diff
@@ -302,6 +302,7 @@ class FieldDictionaryAttributes:
     V = "/V"  # text string, optional
     DV = "/DV"  # text string, optional
     AA = "/AA"  # dictionary, optional
+    Opt = "/Opt" # Options, Optional
```
Comment (MartinThoma) [inline PyPDF2/constants.py:305]: This attribute is not defined in Table 8.69 but in Table 8.76. Please use

```suggestion
class CheckboxRadioButtonAttributes:
    """TABLE 8.76 Additional entry specific to check box and radio button fields."""
    Opt = "/Opt" # Options, Optional
```

The main reason for using those constants is to point to the right part of the documentation / adding additional information.

**Comment by MartinThoma** [comment]
I think I need to check where / how we use that dictionary to understand why you've added it. Could you point me to the place in code where adding the constant improves PyPDF2?

File: PyPDF2/constants.py (hunk 1 above)
Comment (KourFrost) [reply PyPDF2/constants.py:307]: where is the table/documentation stored at?

**Comment by KourFrost** [comment]
https://pypdf2.readthedocs.io/en/latest/_modules/PyPDF2/_reader.html#PdfReader.get_fields

reader.get_fields()  uses that constants list to add information to the dictionary and only returns attributes from that list. 
`field_attributes = FieldDictionaryAttributes.attributes_dict()`

by adding /Opt to that list its now included, and with my limited testing only shows up if the field has Opt attribute. It didn't break any of your other tests in pytest, so I figured that would be a good solution. 

another solution could be adding like an overloaded function to get_fields that takes an additional list of attributes, Then it wouldn't need to update the constant attributes list.

if one would want to see the items in a drop down/combo box you would need something like this:
```
for annot in reader.pages[0]["/Annots"]:
    obj = annot.get_object()
    if "/T" in obj.keys():
        if obj["/T"] == objectTitle:
            if "/Opt" in obj.keys():
                print(obj["/Opt"])
```

**Comment by MartinThoma** [comment]
Thank you for explaining it :hugs: 

Would you mind doing the following?

1. `constants.py`: Add a `attributes` and a `attributes_dict` method to `CheckboxRadioButtonAttributes` similar to `FieldDictionaryAttributes`. Yes, it would be simpler to add this single attribute directly to `FieldDictionaryAttributes`, but that would mean we mix two different things. The `constants.py` should be very close to various tables of the PDF specs.
2. `constants.py`: In the very bottom, there is a `PDF_KEYS`. Please add the `CheckboxRadioButtonAttributes` there. It ensures in the tests that there are no obvious typos.
4. `_reader.py`: Please change `field_attributes = FieldDictionaryAttributes.attributes_dict()` to `field_attributes = FieldDictionaryAttributes.attributes_dict();field_attributes.update(CheckboxRadioButtonAttributes.attributes_dict())`
3. `_reader.py`: Please change `for attr in FieldDictionaryAttributes.attributes():` in a similar way so that we iterate over both
5. Apply [`black .`](https://pypi.org/project/black/) to format the code. You might need to `pip install black` first.

I think that should do the trick while maintaining that `constants.py` represents various tables from the PDF specification. What do you think?

**Comment by KourFrost** [comment]
I like that as a solution. I'll get that implemented.

**Comment by KourFrost** [comment]
It didn't turn out to be as simple as we had hoped. Rewriting get_fields also required a similar change in _write_field and the field object it self. There might be some refactoring that we could do as get_fields loops through the same attributes as _write_field. _write_field is only called in _build_field. _build_field is only called by get_fields. 

Also an issue with fields as a definition. Increasing the scope of what attributes define a field ended being the final trick. Fields objects define themselves off of the FieldDictionaryAttributes constants.

**Comment by KourFrost** [comment]
its all good, stay safe man!

File: PyPDF2/_reader.py [hunk 2]
```diff
@@ -497,6 +501,7 @@ def get_fields(
                 break
 
         if "/Fields" in tree:
+
```
Comment (MartinThoma) [inline PyPDF2/_reader.py:504]:
```suggestion
```

**Review by MartinThoma — APPROVED** [review]
Thank you for the contribution :hugs:

**Comment by MartinThoma** [comment]
Very nice work @KourFrost ! I will make a release of PyPDF2 to PyPI today that contains your contribution!

If you want, I can also add you to the https://pypdf2.readthedocs.io/en/latest/meta/CONTRIBUTORS.html list - Just send me the line I should add for you :-)

**Comment by KourFrost** [comment]
Thank you! This is my first real contribution to an open source project, so I'm happy to help. 

sure add me to the contributors list. just "KourFrost" linked to my github 
https://github.com/KourFrost

**Pull request merged**